dev-tools/omdb/src/bin/omdb/db/sitrep.rs

```diff
@@ -138,53 +138,58 @@ pub(super) async fn cmd_db_sitrep_history(
     struct SitrepRow {
         v: u32,
         id: Uuid,
-        #[tabled(display_with = "datetime_rfc3339_concise")]
-        created_at: DateTime<Utc>,
+        #[tabled(display_with = "display_option_blank")]
+        orphans: Option<usize>,
+        #[tabled(display_with = "datetime_opt_rfc3339_concise")]
+        created_at: Option<DateTime<Utc>>,
         comment: String,
     }
 
-    let conn = datastore.pool_connection_for_tests().await?;
     let marker = args.from.map(model::SqlU32::new);
     let pagparams = DataPageParams {
         marker: marker.as_ref(),
         direction: PaginationOrder::Descending,
         limit: fetch_opts.fetch_limit,
     };
-    let sitreps: Vec<(model::SitrepVersion, model::SitrepMetadata)> =
-        paginated(
-            history_dsl::fm_sitrep_history,
-            history_dsl::version,
-            &pagparams,
-        )
-        .inner_join(
-            sitrep_dsl::fm_sitrep.on(history_dsl::sitrep_id.eq(sitrep_dsl::id)),
-        )
-        .select((
-            model::SitrepVersion::as_select(),
-            model::SitrepMetadata::as_select(),
-        ))
-        .load_async(&*conn)
+    let versions = datastore
+        .fm_sitrep_version_list(&opctx, &pagparams)
         .await
         .with_context(ctx)?;
 
-    check_limit(&sitreps, fetch_opts.fetch_limit, ctx);
-
-    let rows = sitreps.into_iter().map(|(version, metadata)| {
-        let model::SitrepMetadata {
-            id,
-            time_created,
-            comment,
-            creator_id: _,
-            parent_sitrep_id: _,
-            inv_collection_id: _,
-        } = metadata;
-        SitrepRow {
-            v: version.version.into(),
-            id: id.into_untyped_uuid(),
+    check_limit(&versions, fetch_opts.fetch_limit, ctx);
+
+    let mut rows = Vec::with_capacity(versions.len());
+    for v in versions {
+        let orphans = match datastore.fm_sitrep_list_orphaned(&opctx, &v).await
+        {
+            Ok(o) => Some(o.len()),
+            Err(e) => {
+                eprintln!(
+                    "failed to list orphaned sitreps at v{}: {e}",
+                    v.version
+                );
+                None
+            }
+        };
+        let (comment, time_created) =
+            match datastore.fm_sitrep_metadata_read(&opctx, v.id).await {
+                Ok(s) => (s.comment, Some(s.time_created)),
+                Err(e) => {
+                    eprintln!(
+                        "failed to get fetch metadata for sitrep {} (v{}): {e}",
+                        v.id, v.version
+                    );
```

Comment:
Would this be the expected case if we did a partial insert of sitrep data, but the "final sitrep" was not inserted?

Or are we saying: "if any records exist correlated with a sitrep, it is guaranteed to have an entry in the  `fm_sitrep` table"?

Reply:
Edit: Seems like you're relying on that ordering, as I'm reading below.

Reply:
bf33ebf1498c687c85eb584cf04933b47fa23ff4 added a quick comment describing this expectation.